modules/light-clients/08-wasm/types/client_state.go

```diff
@@ -250,17 +257,37 @@ func call[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *Client
 	var output T
 	encodedData, err := json.Marshal(payload)
 	if err != nil {
-		return output, sdkerrors.Wrapf(err, "failed to marshal wasm contract payload")
+		return output, sdkerrors.Wrapf(err, "failed to marshal payload for wasm execution")
 	}
 	out, err := callContract(ctx, clientStore, cs.CodeId, encodedData)
 	if err != nil {
 		return output, sdkerrors.Wrapf(err, "call to wasm contract failed")
 	}
 	if err := json.Unmarshal(out.Data, &output); err != nil {
-		return output, sdkerrors.Wrapf(err, "failed unmarshal wasm contract payload")
+		return output, sdkerrors.Wrapf(err, "failed to unmarshal result of wasm execution")
+	}
+	if !output.Validate() {
+		return output, sdkerrors.Wrapf(errors.New(output.Error()), "error occurred while executing contract with code ID %s", hex.EncodeToString(cs.CodeId))
+	}
+	return output, nil
+}
+
+// wasmQuery queries the contract with the given payload and writes the result to output.
+func wasmQuery[T ContractResult](ctx sdk.Context, clientStore sdk.KVStore, cs *ClientState, payload any) (T, error) {
```

Comment:
I added this function (which is very similar to `call` above) to wrap the query call to the contract. Since the query (still to be added) to return the timestamp at a certain height (with payload of type `timestampAtHeightQueryResponse`) might return an error if there's no consensus at that given height, I thought that we can also check the result and wrap if the error if there's any.

Reply:
I had to named this function like this and not `query`, because `query` conflicted with another function defined in a package in the SDK that is also in scope. I will probably rename `call` above to `wasmExecute`, for consistency.